### PR TITLE
[kitchen] Add tests to check that the Agent does start with Python 3

### PR DESCRIPTION
**Pull request by KSerrania**
### What does this PR do?

Adds sanity checks to verify that we do include Python 3 and that the Agent can restart in Python 2 or Python 3 mode.

### Motivation

Python 3 support starting with `6.14.0`.

### Additional Notes

Kitchen tests currently fail because the install directory is not removed on Linux. See the related [datadog-agent PR](https://github.com/DataDog/datadog-agent/pull/4032) and [integrations-core PR](https://github.com/DataDog/integrations-core/pull/4371).

### CODE REVIEW

File: test/kitchen/test/integration/common/rspec/spec_helper.rb
Comment (truthbk) [inline test/kitchen/test/integration/common/rspec/spec_helper.rb]: Though this is enough to check the agent comes up, I'm wondering if we shouldn't check the `info` page (status), to actually verify the correct interpreter is being used. Feel free to disregard if you think that's beyond the scope of what we're testing.

**Review by truthbk — COMMENTED** [review]
Looks good, just added a comment regarding verification of what is actually running.

File: test/kitchen/test/integration/common/rspec/spec_helper.rb
Comment (olivielpeau) [inline test/kitchen/test/integration/common/rspec/spec_helper.rb]: note: I don't know how many times you've tried to run the kitchen tests with this new test, but I think it may take some time for the Agent to be fully initialized (including having loaded the python interpreter) and actually expose `pythonV` in expvar.

So I think this test has chances of being a bit flaky if for some reason the Agent takes some time to initialize. Ideally there would be some logic here to wait until the expvar key is actually present (and timeout after x seconds).

No need to address this in this PR, but please be on the lookout for kitchen flakiness once this is merged, and think about this potential edge case if you see flakiness :)

**Review by olivielpeau — DISMISSED** [review]
Just one comment that can be addressed later. LGTM!

File: test/kitchen/test/integration/common/rspec/spec_helper.rb
Comment (KSerrania) [reply test/kitchen/test/integration/common/rspec/spec_helper.rb]: Actually I need to address this, I've tested 4 times, and each time it has failed at least once, so I'll add the timeout logic

**Review by truthbk — DISMISSED** [review]
👌 🚀

**Review by truthbk — DISMISSED** [review]
Approving again - if this is now consistently succeeding feel free to merge.

**Comment by KSerrania** [comment]
Tests are still flaky 😞 I'll revert the "check `datadog-agent status`" part for now and merge the "start with python3" checks only, and I'll get back to this after the freeze

**Review by albertvaka — APPROVED** [review]

**Pull request merged**
